content/Watsonx/WatsonxAI/102.md

```diff
@@ -0,0 +1,129 @@
+---
+title: '102: Programmatic Access'
+timeToComplete: 30
+updated: 2023-09-19
+---
+
+# 102: Programmatic Access to watsonx.ai
+
+In this lab we'll explore two ways that we can programmatically access watsonx.ai programmatically. We will do this through the use of some open source tools like Python and Jupyter Notebooks.
+
+First though, we'll need to acquire some prerequisites from your IBM Cloud account in order to access the APIs
+
+## Getting prerequisites
+
+### Creating an API Key
+
+In order to authenticate with watsonx externally via an API you will first need to create an API key that's tied to your IBM Cloud Account.
+
+To do this:
+
+1. Start by heading to your [cloud account's dashboard](https://cloud.ibm.com/)
+2. Click **Manage** at the top of the page, then **Access (IAM)**
+
+![to_iam](./images/102/1.png)
+
+3. In the left-hand navigation menu click **API Keys**
+
+![iam_page](./images/102/2.png)
+
+4. A table should appear with a list of your current API keys. Click the **Create +** button on the top-right of the table
+
+![api_keys](./images/102/3.png)
+
+5. In the menu that appears given your API key a name and an optional description and click the **Create** button
+
+![create_key](./images/102/4.png)
+
+6. Lastly, A new menu will appear titled **API key successfully created**. Click either the **Copy** or the **Download** button. If clicking copy be sure to immediately store this value in a text file for later use.
+
+![copy_key](./images/102/5.png)
+
+With that you have a valid API key.
+
+### Getting your project ID
+
+We now need to find the project ID of your watsonx instance. Start by navigating to the overview page for watsonx [here](https://cloud.ibm.com/watsonx/overview). Once there:
+
+1. Click the **Launch** button inside the **watsonx.ai** card on the page. This will open a new tab.
+
+![launch_ai](./images/102/6.png)
```

Comment:
Maybe it's just me but the button says `Get started` instead of `Launch`

<img width="1728" alt="Screenshot 2023-09-22 at 1 44 16 PM" src="https://github.com/ibm-build-lab/VAD-VAR-Workshop/assets/19367627/8306b82e-3fc0-4524-94d7-73d92e89d52a">

Reply:
I previously had my watsonx instance set up. Sometimes I see `get started` flash but I'm not sure what could cause the difference otherwise.

Reply:
I believe it depends on the IBM cloud account that you currently have selected.  For instance, navigating to this page with a cloud account that does not have watsonx services provisioned I believe would show "Getting started".  See the two screenshots, one with Tim's account and one with a random RHM account.  Can we add a note to clarify?
 
> If you see "Get started" rather than "Launch", make sure you have selected an IBM cloud account (from the dropdown in the top right) where watsonx services are deployed.

![Screen Shot 2023-09-26 at 10 42 21 AM](https://github.com/ibm-build-lab/VAD-VAR-Workshop/assets/7364498/5e9c463c-2ffb-4a70-b027-73b9ad98b449)
![Screen Shot 2023-09-26 at 10 42 43 AM](https://github.com/ibm-build-lab/VAD-VAR-Workshop/assets/7364498/6a3f7e3b-a08c-4f3e-abfc-8d2cf52a471b)